src/app/project/project-permissions/permissions-table-rows/permissions-table-rows.component.ts

```diff
@@ -39,36 +46,100 @@ export class PermissionsTableRowsComponent implements OnInit, TableComponent {
    *
    * @return {void}
    */
-  ngOnInit() {
+  ngOnInit(): void {
     this.currentProject = this.storageService.state.currentProject.data;
     this.entries = this.data.data;
+    this.paginationData = this.data.paginationData;
+    this.route.params
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe(params => {
+        this.tableParams = this.tableTemplateUtils.getParamsFromUrl(params);
+      });
   }
 
   /**
    * Opens a new snack bar notification message with a duration of 2 seconds, and executes an action.
    *
    * @param {string} message A snack bar notification message.
    * @param {string} action A snack bar notification action.
-   * @returns {void}
+   * @return {void}
    */
    public openSnackBar(message: string, action: string): void {
     this.snackBar.open(message, action, {
       duration: 2000,
     });
   }
 
-  hasProjectPermission(user) {
+  /**
+   * Determines if a user has project permission.
+   *
+   * @param {User} user The user to check.
+   * @returns {boolean}
+   */
+  hasProjectPermission(user: User): boolean {
     return user.projectPermissions.includes(this.currentProject._id);
   }
 
+  /**
+   * Removes duplicate user entries.
+   *
+   * @param {User[]} returnedUsers The user list that will be checked for duplicate users.
+   * @returns {User[]}
+   */
+  removeDuplicateUsers(returnedUsers: User[]): User[] {
+    let userNames = [];
```

Comment:
```suggestion
    const userNames = [];
    const validatedUsers = [];
```
Mutable types like arrays and objects can be assigned as constants and still modified. You could use `const` here for a bit more safety and not run into any issues